lib/internal/Magento/Framework/CompiledInterception/Generator/CompiledInterceptor.php

```diff
@@ -0,0 +1,532 @@
+    {
+        $ret = [];
+        foreach ($parameters as $parameter) {
+            $ret [] = "\${$parameter->getName()}";
+        }
+        return implode(', ', $ret);
+    }
+
+    /**
+     * @param $plugin
+     * @return string
+     */
+    private function getGetterName($plugin)
+    {
+        return '____plugin_' . $plugin['clean_name'];
+    }
+
+    /**
+     * @param $plugin
+     * @return array
+     */
+    private function getPluginPropertyInfo($plugin)
+    {
+        return [
+            'name' => '____plugin_' . $plugin['clean_name'],
+            'visibility' => 'private',
+            'docblock' => [
+                'tags' => [['name' => 'var', 'description' => '\\' . $plugin['class']]],
+            ]
+        ];
+    }
+
+    /**
+     * @param $plugin
+     * @return array
+     */
+    private function getPluginGetterInfo($plugin)
+    {
+        $body = [];
+        $varName = "\$this->____plugin_" . $plugin['clean_name'];
+
+        $body[] = "if ($varName === null) {";
+        $body[] = "\t$varName = \$this->____om->get(\\" . "{$plugin['class']}::class);";
+        $body[] = "}";
+        $body[] = "return $varName;";
+
+        return [
+            'name' => $this->getGetterName($plugin),
+            'parameters' => [],
+            'body' => implode("\n", $body),
+            'returnType' => $plugin['class'],
+            'docblock' => [
+                'shortDescription' => 'plugin "' . $plugin['code'] . '"' . "\n" . '@return \\' . $plugin['class']
+            ],
+        ];
+    }
+
+    /**
+     * @param \ReflectionMethod $method
+     * @param $config
+     * @return array
+     */
+    private function getCompiledMethodInfo(\ReflectionMethod $method, $config)
+    {
+        $parameters = $method->getParameters();
+        $returnsVoid = ($method->hasReturnType() && $method->getReturnType()->getName() == 'void');
+
+        $body = [
+            'switch($this->____scope->getCurrentScope()){'
+        ];
+
+        $cases = [];
+        //group cases by config
+        foreach ($config as $scope => $conf) {
+            $key = md5(serialize($conf));
+            if (!isset($cases[$key])) $cases[$key] = ['cases'=>[], 'conf'=>$conf];
+            $cases[$key]['cases'][] = "\tcase '$scope':";
+        }
+        //call parent method for scopes with no plugins (or when no scope is set)
+        $cases[] = ['cases'=>["\tdefault:"], 'conf'=>[]];
+
+        foreach ($cases as $case) {
+            $body = array_merge($body, $case['cases']);
+            $this->addCodeSubBlock($body, $this->getMethodSourceFromConfig($method->getName(), $case['conf'], $parameters, $returnsVoid), 2);
+            if ($returnsVoid) {
+                $body[] = "\t\tbreak;";
+            }
+        }
+
+        $body[] = "}";
+        $returnType = $method->getReturnType();
+        $returnTypeValue = $returnType
+            ? ($returnType->allowsNull() ? '?' : '') .$returnType->getName()
```

Comment:
Should be whitespace between `.` and `$returnType`

Reply:
fixed